pandas/_libs/tslibs/src/datetime/np_datetime_strings.c

```diff
@@ -139,6 +133,9 @@ int parse_iso_8601_datetime(const char *str, int len, int want_exc,
     while (sublen > 0 && isspace(*substr)) {
         ++substr;
         --sublen;
+        if (exact == PARTIAL_MATCH && !format_len) {
```

Comment:
Can we not just make `compare_format` return a set of Enum depending on what is left in the string to consume and what the matching semantics are? Seems like it would naturally fit there rather than a separate branch every time

Reply:
To clarify, I think you can return an enum from `check_format` of values like:

```sh
OK_EXACT
OK_PARTIAL
```

etc... describing the different states, then branch in the caller appropriately